src/managedwriter/stream_connection.ts

```diff
@@ -280,16 +324,19 @@ export class StreamConnection extends EventEmitter {
 
   private send(pw: PendingWrite) {
     const request = pw.getRequest();
-    if (!this._connection) {
-      pw._markDone(new Error('connection closed'));
-      return;
-    }
-    if (this._connection.destroyed || this._connection.closed) {
+    if (this.isConnectionClosed()) {
       this.reconnect();
     }
     this.trace('sending pending write', pw);
+    const retrySettings = this._writeClient['_retrySettings'];
+    const tries = pw._increaseRetryAttempts();
```

Comment:
move this before reconnect?

Reply:
If maxRetryAttempts is 0, does it mean we should send the request? So it should be tries >= retrySettings.maxRetryAttempts?

Reply:
if we change to `>=`, on the first time the request is sent ( with no retries yet ) it is going to throw a error. So I think it should still be just `>`.

Reply:
Resolve.